backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/util/AsyncAggregatingSubscriber.java

```diff
@@ -85,10 +93,12 @@ public void onComplete() {
    */
   public CompletableFuture<Collection<T>> result() {
     return CompletableFuture.supplyAsync(phaser::arriveAndAwaitAdvance)
-        .thenApply(
+        .thenCompose(
             ignored -> {
-              LOG.trace("Result is available: {}", results);
-              return results;
+              if (!resultsFuture.isDone()) {
```

Comment:
:wrench: no need to check here, `complete` only does something if the future is not yet done.